packages/app/src/lib/import.ts

```diff
@@ -244,9 +251,113 @@ export async function asLastPass(data: string): Promise<VaultItem[]> {
+    } catch (error) {
+        return false;
+    }
 }
 
-export function guessFormat(data: string): ImportFormat | null {
-    return isPBES2Container(data) ? PBES2 : isPadlockV1(data) ? PADLOCK_LEGACY : isLastPass(data) ? LASTPASS : CSV;
+async function parse1PuxItem(accountName: string, vaultName: string, item: OnePuxItem['item']): Promise<VaultItem | undefined> {
+    if (!item) {
+        return;
+    }
+
+    const rowData = parseToRowData(item, [accountName, vaultName]);
+
+    if (!rowData) {
+        return;
+    }
+
+    const itemName = rowData.name;
+    const tags = rowData.tags.split(',');
+
+    if (item.trashed) {
+        tags.push('trashed');
+    }
+
+    let fields: Field[] = [
+        new Field({ name: $l("Username"), value: rowData.username, type: FieldType.Username }),
+        new Field({ name: $l("Password"), value: rowData.password, type: FieldType.Password }),
+        new Field({ name: $l("URL"), value: rowData.url, type: FieldType.Url }),
+    ];
+
+    if (rowData.notes) {
+        fields.push(new Field({ name: $l("Notes"), value: rowData.notes, type: FieldType.Note }));
+    }
+
+    for (const extraField of rowData.extraFields) {
+        if (extraField.type === 'totp') {
+            // Extract just the secret
+            try {
+                const secret = new URL(extraField.value).searchParams.get('secret');
+                if (secret) {
+                    fields.push(new Field({ name: extraField.name, value: secret, type: FieldType.Totp }));
+                }
+            } catch (error) {
+                // Do nothing
+            }
+        } else {
+            // @ts-ignore All of extraField.type possibilities match FieldType.*
+            fields.push(new Field({ name: extraField.name, value: extraField.value, type: extraField.type }));
+        }
+
+    }
+
+    return createVaultItem(itemName, fields, tags);
+}
+
+export async function as1Pux(file: string | Uint8Array): Promise<VaultItem[]> {
+    try {
+        const dataExport = await parse1PuxFile(file);
+
+        const items = [];
+        
+        for (const account of dataExport.data.accounts) {
+            for (const vault of account.vaults) {
+                for (const vaultItem of vault.items) {
+                    if (vaultItem.item) {
+                        const parsedItem = await parse1PuxItem(account.attrs.name, vault.attrs.name, vaultItem.item);
+                        if (parsedItem) {
+                            items.push(parsedItem);
+                        }
+                    }
+                }
+            }
+        }
+
+        return items;
+    } catch (error) {
+        throw new Err(ErrorCode.INVALID_1PUX);
+    }
+}
+
+/**
+ * Checks if a given string/Uint8Array represents a 1Password 1pux file
+ */
+export async function is1Pux(file: string | Uint8Array): Promise<boolean> {
+    try {
+        const dataExport = await parse1PuxFile(file);
```

Comment:
I'm thinking it would probably be easer if we simply check the file extension. For that we'd need a reference to the file object of course, but I'm thinking maybe it would be better to keep a reference of the File object instead of `string | Uint8Array` anyway? The file would then only be read / parsed when necessary. The big advantage of this would be that we could load the relevant code on demand, instead of having to load it for every file type.

Reply:
That makes sense. I'll make that change.